From a45447eac1cb839b7c4429811590a46272e7525e Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd@kuhls.net>
Date: Sun, 20 Sep 2026 15:01:10 +0200
Subject: [PATCH] ssl: Fix remaining build errors with OpenSSL 4.x

ssl/Security.cxx: In static member function 'static void resip::BaseSecurity::getCertNames(X509*, std::__cxx11::list<PeerName>&, bool)':
ssl/Security.cxx:2583:51: error: invalid conversion from 'const X509_NAME_ENTRY*' {aka 'const X509_name_entry_st*'} to 'X509_NAME_ENTRY*' {aka 'X509_name_entry_st*'} [-fpermissive]
 2583 |       X509_NAME_ENTRY* entry = X509_NAME_get_entry(subject,i);
      |                                ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
      |                                                   |
      |                                                   const X509_NAME_ENTRY* {aka const X509_name_entry_st*}

ssl/Security.cxx:2586:53: error: invalid conversion from 'const ASN1_STRING*' {aka 'const asn1_string_st*'} to 'ASN1_STRING*' {aka 'asn1_string_st*'} [-fpermissive]
 2586 |       ASN1_STRING*      s = X509_NAME_ENTRY_get_data(entry);
      |                             ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
      |                                                     |
      |                                                     const ASN1_STRING* {aka const asn1_string_st*}

Fixes https://github.com/resiprocate/resiprocate/issues/489

Upstream: https://github.com/resiprocate/resiprocate/commit/45729f0f8d34ef5c819b3827c8334546318d1c49

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 resip/stack/ssl/Security.cxx | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/resip/stack/ssl/Security.cxx b/resip/stack/ssl/Security.cxx
index 563700cca..28348c9ce 100644
--- a/resip/stack/ssl/Security.cxx
+++ b/resip/stack/ssl/Security.cxx
@@ -2616,10 +2616,10 @@ BaseSecurity::getCertNames(X509 *cert, std::list<PeerName> &peerNames,
          break;
       }
       resip_assert( i != -1 );
-      X509_NAME_ENTRY* entry = X509_NAME_get_entry(subject,i);
+      const X509_NAME_ENTRY* entry = X509_NAME_get_entry(subject,i);
       resip_assert( entry );
       
-      ASN1_STRING*	s = X509_NAME_ENTRY_get_data(entry);
+      const ASN1_STRING*	s = X509_NAME_ENTRY_get_data(entry);
       resip_assert( s );
       
       int t = ASN1_STRING_type(s);
-- 
2.47.3

